src/common/traffic-source.js

```diff
@@ -53,14 +61,16 @@ const TrafficSource = (() => {
     const setData = () => {
         const params = queryString.parseUrl(window.location.href).query
         const param_keys = ['utm_source', 'utm_medium', 'utm_campaign']
-
-        if (params.utm_source) {
-            // url params can be stored only if utm_source is available
-            param_keys.forEach(key => {
-                if (params[key]) {
-                    cookie.set(key, params[key], { sameSite: 'none', secure: true })
-                }
-            })
+        const makeSetCookie = (object) => (key) =>
+            object[key] && cookie.set(key, object[key], { sameSite: 'none', secure: true })
+
+        if (params.utm_source || params.utm_medium || params.utm_campaign) {
+            const setParamCookie = makeSetCookie(params)
+            param_keys.forEach(setParamCookie)
+        } else if (existing_utm_cookie) {
```

Comment:
if it only has affiliate token without utms, then it should have been no utm right?, can we check both if there is affiliate token and utm in the param, if not then we can use the old one

Reply:
Do you mean: if the url only has affiliate token -- but no utm params -- we should not use the old cookie?

Reply:
thats right